src/core/qgsvectorlayerutils.cpp

```diff
@@ -320,3 +323,92 @@ QgsFeature QgsVectorLayerUtils::createFeature( QgsVectorLayer *layer, const QgsG
   return newFeature;
 }
 
+QgsFeature QgsVectorLayerUtils::duplicateFeature( QgsVectorLayer *layer, const QgsFeature &feature, QgsProject *project, int depth, QgsDuplicateFeatureContext &duplicateFeatureContext )
+{
+  if ( !layer )
+    return QgsFeature();
+
+  if ( !layer->isEditable() )
+    return QgsFeature();
+
+  //get context from layer
+  QgsExpressionContext context = layer->createExpressionContext();
+  context.setFeature( feature );
+
+  //create the attribute map
+  QgsAttributes srcAttr = feature.attributes();
+  QgsAttributeMap dstAttr;
+  for ( int src = 0; src < srcAttr.count(); ++src )
+  {
+    dstAttr[ src ] = srcAttr.at( src );
+  }
+
+  QgsFeature newFeature = createFeature( layer, feature.geometry(), dstAttr, &context );
+
+  const QList<QgsRelation> relations = project->relationManager()->referencedRelations( layer );
+
+  QgsDebugMsg( QStringLiteral( "check relations" ) );
```

Comment:
Remove all these `QgsDebugMsg` calls (except if there are some which you expect to be useful for debugging in future)